test/unit/sessions/test_attachment_models.py

```diff
@@ -0,0 +1,518 @@
+            rootPathFormat=PathFormat.POSIX,
+            fileSystemLocationName="shared",
+        )
+
+        manifest_props2 = ManifestProperties(
+            rootPath="/source/path",
+            rootPathFormat=PathFormat.POSIX,
+            fileSystemLocationName="shared",
+        )
+
+        worker_props1 = WorkerManifestProperties(
+            manifest_properties=manifest_props1,
+            local_root_path="/local/root",
+            local_manifest_paths=["/local/manifest.json"],
+        )
+
+        worker_props2 = WorkerManifestProperties(
+            manifest_properties=manifest_props2,
+            local_root_path="/local/root",
+            local_manifest_paths=["/local/manifest.json"],
+        )
+
+        worker_props3 = WorkerManifestProperties(
+            manifest_properties=manifest_props1,
+            local_root_path="/different/root",
+            local_manifest_paths=["/local/manifest.json"],
+        )
+
+        # Test equality (dataclass should provide __eq__)
+        assert worker_props1 == worker_props2
+        assert worker_props1 != worker_props3
+
+    def test_from_manifest_properties_with_class_method_validation(self):
+        """Test that class method properly validates inputs."""
+        manifest_props = ManifestProperties(
+            rootPath="/source/path", rootPathFormat=PathFormat.POSIX
+        )
+
+        # Test with valid inputs
+        worker_props = WorkerManifestProperties.from_manifest_properties(
+            manifest_properties=manifest_props, local_root_path="/valid/root"
+        )
+        assert worker_props.local_root_path == "/valid/root"
+
+        # Test that validation still occurs in class method
+        with pytest.raises(ValueError, match="local_root_path must be set"):
+            WorkerManifestProperties.from_manifest_properties(
+                manifest_properties=manifest_props, local_root_path=""
+            )
+
+    def test_local_manifest_paths_default_factory(self):
+        """Test that each instance gets its own local_manifest_paths list (mutable default fix)."""
+        manifest_props1 = ManifestProperties(
+            rootPath="/source/path1", rootPathFormat=PathFormat.POSIX
+        )
+        manifest_props2 = ManifestProperties(
+            rootPath="/source/path2", rootPathFormat=PathFormat.POSIX
+        )
+
+        # Create two instances without specifying local_manifest_paths
+        worker_props1 = WorkerManifestProperties(
+            manifest_properties=manifest_props1, local_root_path="/local/root1"
+        )
+        worker_props2 = WorkerManifestProperties(
+            manifest_properties=manifest_props2, local_root_path="/local/root2"
+        )
+
+        # Verify each instance has its own empty list
+        assert worker_props1.local_manifest_paths == []
+        assert worker_props2.local_manifest_paths == []
+        assert worker_props1.local_manifest_paths is not worker_props2.local_manifest_paths
```

Comment:
This is a good test to have. It prevents a common Python pitfall where the default is a reference to a single list